### PR TITLE
ConvectiveHeatFluxBC

### LINKED ISSUE #14814: ConvectiveHeatFluxBC with material properties
## Reason
There are several convective BC, but none use material properties for the heat transfer coefficient or far field temperature. Such a BC is useful when material properties are used instead of variables.

## Design
Use most of `CoupledConvectiveHeatFluxBC`, but couple materials instead of variables.

## Impact
Increased BC options.

### PR DESCRIPTION
**Pull request by tophmatthews**
Closes #14814

### CODE REVIEW

**Comment by tophmatthews** [comment]
maybe... @andrsd since you did similar things?

File: modules/heat_conduction/include/bcs/ConvectiveHeatFluxBC.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+public:
+  static InputParameters validParams();
+
+  ConvectiveHeatFluxBC(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual() override;
+  virtual Real computeQpJacobian() override;
+
+  /// Far-field temperatue fields for each component
```
Comment (andrsd) [inline modules/heat_conduction/include/bcs/ConvectiveHeatFluxBC.h]: Just "Far-field temperature field", i.e. (a) fix the typo and (b) you are dealing with just one field.

File: modules/heat_conduction/test/tests/convective_heat_flux/flux.i [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# on the left side and a convective flux BC on the right side.
+# The temperature on the left is 100, and the far-field temp is 200.
+# The conductance of the body (conductivity * length) is 10
+#
+# If the conductance in the BC is also 10, the temperature on the
+# right side of the solid element should be 150 because half of the
+# temperature drop should occur over the body and half in the BC.
+#
+# The integrated flux is deltaT * conductance, or -50 * 10 = -500.
+# The negative sign indicates that heat is going into the body.
```
Comment (andrsd) [inline modules/heat_conduction/test/tests/convective_heat_flux/flux.i]: 👍

File: modules/heat_conduction/test/tests/convective_heat_flux/tests [hunk 3]
```diff
@@ -0,0 +1,25 @@
+[Tests]
+  issues = '#11631'
+  design = 'ConvectiveHeatFluxBC.md'
+  [./g]
+  requirement = 'The system shall provide a convective flux boundary condition which uses material properties as heat transfer coefficients and far-field temperature values'
```
Comment (andrsd) [inline modules/heat_conduction/test/tests/convective_heat_flux/tests]: Indent the requirement by 2 space

File: modules/heat_conduction/include/bcs/ADConvectiveHeatFluxBC.h [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+{
+public:
+  static InputParameters validParams();
+
+  ADConvectiveHeatFluxBC(const InputParameters & parameters);
+
+protected:
+  virtual ADReal computeQpResidual() override;
+
+  /// Far-field temperatue fields for each component
```
Comment (andrsd) [inline modules/heat_conduction/include/bcs/ADConvectiveHeatFluxBC.h]: "Far-field temperature field" only.

File: modules/heat_conduction/test/tests/ad_convective_heat_flux/tests [hunk 5]
```diff
@@ -0,0 +1,25 @@
+[Tests]
+  issues = '#11631'
+  design = 'ADConvectiveHeatFluxBC.md'
+  [./g]
+  requirement = 'The system shall provide a convective flux boundary condition which uses material properties as heat transfer coefficients and far-field temperature values using AD'
```
Comment (andrsd) [inline modules/heat_conduction/test/tests/ad_convective_heat_flux/tests]: Indent by 2 spaces

**Review by andrsd — CHANGES_REQUESTED** [review]

File: modules/heat_conduction/include/bcs/ADConvectiveHeatFluxBC.h [hunk 6]
```diff
@@ -31,7 +31,7 @@ class ADConvectiveHeatFluxBC : public ADIntegratedBC<compute_stage>
 protected:
   virtual ADReal computeQpResidual() override;
 
-  /// Far-field temperatue fields for each component
+  /// Far-field temperatue variable
```
Comment (andrsd) [inline modules/heat_conduction/include/bcs/ADConvectiveHeatFluxBC.h]: temperatue -> temperatu**R**e :-) You missed that one :-)

File: modules/heat_conduction/include/bcs/ConvectiveHeatFluxBC.h [hunk 7]
```diff
@@ -31,7 +31,7 @@ class ConvectiveHeatFluxBC : public IntegratedBC
   virtual Real computeQpResidual() override;
   virtual Real computeQpJacobian() override;
 
-  /// Far-field temperatue fields for each component
+  /// Far-field temperatue variable
```
Comment (andrsd) [inline modules/heat_conduction/include/bcs/ConvectiveHeatFluxBC.h]: temperatue -> temperatu**R**e :-) And here as well :-)

**Review by andrsd — CHANGES_REQUESTED** [review]

File: modules/heat_conduction/include/bcs/ADConvectiveHeatFluxBC.h (hunk 6 above)
Comment (tophmatthews) [reply modules/heat_conduction/include/bcs/ADConvectiveHeatFluxBC.h]: ug....

**Comment by tophmatthews** [comment]
yeah...poor copy paste from `CoupledConvectiveHeatFluxBC.h `. There were a couple other `temperatue`s I corrected here.

**Review by andrsd — APPROVED** [review]

**Pull request merged**